src/Processor/Braintree.php

```diff
@@ -331,6 +340,52 @@ public function updateSubscription(Subscription $subscription): Subscription
         }
     }
 
+    private function changeSubscriptionBillingCycle(Subscription $subscription): Subscription
+    {
+        $customer = $subscription->getCustomer();
+        $newPlan = $subscription->getPlan();
+        $newPrice = $subscription->getPrice();
+        $newAddOns = $subscription->getAddOns();
+        $newDiscounts = $subscription->getDiscounts();
+
+        if (!isset($newPlan, $customer)) {
+            throw new SubscriptionNotUpdatedException('Cannot change subscription billing cycle with null plan or customer');
+        }
+
+        // Zero out subscription to get a balance
+        $subscription->closeOut();
+        $result = $this->doBraintreeUpdate($subscription);
+        if (!$result->success) {
+            $message = isset($result->message) ? $result->message : 'An unknown update error occurred';
+            throw new SubscriptionNotUpdatedException($message);
+        }
+        $updated = $this->findSubscriptionById((string) $subscription->getId());
```

Comment:
This is a pretty interesting exercise in using PHP 7.4 with static analysis. I think things are established enough to say all ids are strings. Maybe `isNew` should just be checking if the string is empty. 

I think something like this:
```php
$this->findSubscriptionById((string) $subscription->getId());
```

Is preferable to this:

```php
$this->findSubscriptionById($subscription->getId() ?? '');
```

But maybe it would be better to just make ids strings and `isNew` will return true if it is an empty string. Would probably make the API simpler. I'll tackle this in a follow-up PR.

Reply:
I think removing nullable types will make the other stuff simpler